MQTTClient/MQTTClient/MQTTSession.m

```diff
@@ -572,24 +574,28 @@ - (void)closeInternal {
                                          code:MQTTSessionErrorNoResponse
                                      userInfo:@{NSLocalizedDescriptionKey : @"No response"}];
 
-    NSArray *allSubscribeHandlers = self.subscribeHandlers.allValues;
-    [self.subscribeHandlers removeAllObjects];
-    for (MQTTSubscribeHandler subscribeHandler in allSubscribeHandlers) {
-        subscribeHandler(error, nil);
+    if(self.subscribeHandlers) {
+        NSArray *allSubscribeHandlers = self.subscribeHandlers.allValues;
+        [self.subscribeHandlers removeAllObjects];
+        for (MQTTSubscribeHandler subscribeHandler in allSubscribeHandlers) {
+            subscribeHandler(error, nil);
+        }
     }
 
-    NSArray *allUnsubscribeHandlers = self.unsubscribeHandlers.allValues;
-    [self.unsubscribeHandlers removeAllObjects];
-    for (MQTTUnsubscribeHandler unsubscribeHandler in allUnsubscribeHandlers) {
-        unsubscribeHandler(error);
+    if(self.unsubscribeHandlers) {
+        NSArray *allUnsubscribeHandlers = self.unsubscribeHandlers.allValues;
+        [self.unsubscribeHandlers removeAllObjects];
+        for (MQTTUnsubscribeHandler unsubscribeHandler in allUnsubscribeHandlers) {
+            unsubscribeHandler(error);
+        }
     }
 
     MQTTDisconnectHandler disconnectHandler = self.disconnectHandler;
     if (disconnectHandler) {
```

Comment:
Why no check here if disconnetHandler is != nil

Reply:
it's irrelevant

Reply:
then if it's irrelevant why do you need to assign self.disconnectHandler = nil outside of this check?